python/TestHarness/TestHarness.py

```diff
@@ -235,87 +242,140 @@ def launchTest(self, find_only, file, dirpath, testers):
+        os.chdir(pbs_job_dir)
+        job_instance = self.factory.create('PBSJob', tester.specs['test_name'], tester.specs)
+        specs = job_instance.specs
+        path_friendly = specs['test_name'].replace('/', '_')
+
+        ## Set some PBS specific specs that help populate the qsub script (and directory)
+        specs['job_name'] = path_friendly
+        specs['test_dir'] = pbs_job_dir
+        specs['template_script'] = self.moose_dir + '/python/TestHarness/pbs_submit.sh'
+        specs['no_copy_pattern'] = '.*\.sh'
+        specs['copy_files'] = 'gold'
+        specs['mpi_procs'] = '1'
+        specs['soft_link_output'] = False
+        specs['place'] = 'scatter:excl'
+        specs['chunks'] = '1'
+        specs['no_copy'] = self.options.input_file_name
+
+        # Convert MAX_TIME to hours:minutes for walltime use
+        hours = int(int(specs['max_time']) / 3600)
+        minutes = int(int(specs['max_time']) / 60) % 60
+        specs['walltime'] = '{0:02d}'.format(hours) + ':' + '{0:02d}'.format(minutes) + ':00'
+
+        # If using the PBS Emulator, set some specifics here
+        if self.options.PBSEmulator:
+            specs['pbs_stdout'] = 'pbs_stdout = PBS_EMULATOR'
+            specs['pbs_stderr'] = 'pbs_stderr = PBS_EMULATOR'
+        # The PBS Emulator fails when using the following
+        else:
+            specs['queue'] = '#PBS -q %s' % (self.options.pbs_project)
+            specs['combine_streams'] = '#PBS -j oe'
+            specs['pbs_project'] = '#PBS -P %s' % (self.options.pbs_project)
 
-        # Close the tests.cluster file
-        if self.options.cluster_handle is not None:
-            self.options.cluster_handle.close()
-            self.options.cluster_handle = None
+        specs.addStringSubParam('command', 'COMMAND', "Command to run")
+        specs['command'] = command
 
-        # Return the final tester/command (sufficient to run all tests)
-        return (tester, command)
+        # Create the PBS Prereq spec. RunParallel will fill in Job ID later. As it is only
+        # after the job is launched, do we know the PBS ID to use here.
+        specs.addStringSubParam('pbs_prereq', 'PBS_PREREQ', "A job dependency that this test requires before running in PBS Dependency syntax")
+        if specs['prereq'] != []:
+            specs['pbs_prereq'] = '#PBS -W depend=afterany:<JOB_ID>'
+
+        job_instance.copyFiles(self.options.input_file_name)
+        job_instance.prepareJobScript()
+
+        # qsub command pointing at our qsub shell script
+        qsub_script = os.path.join(pbs_job_dir, self.options.pbs + '-' + path_friendly + '.sh')
+
+        # So we can support multiple pbs batch scripts in the same directory
+        os.rename(os.path.join(pbs_job_dir, os.path.split(specs['template_script'])[1]),
+                  os.path.join(qsub_script))
+
+        command = 'qsub ' + qsub_script
+
+        # Set the tester status
+        tester.setStatus('LAUNCHED', tester.bucket_pbs)
```

Comment:
Remove this line. The test status is updated with the same status a second time. But with the addition of a Job ID.